### PR TITLE
Verify that the version read from module with WMI command matches the…

### PR DESCRIPTION
**Pull request by Mikolaj-Filar-Mobica**
… version of uploaded image

This pull request cover:
- checking the version of the uploaded image and comparing it with the one read from the module using WMI command
- missing ifdefs in headers
- missed mutex unlock in `athn_usb_htc_setup`

This PR needs to be thoroughly reviewed by Norbert, as it's unknown why the OpenBSD code needed these changes. Since this PR isn't critical now, it can wait for someone with deeper knowledge of USB.

### CODE REVIEW

File: sys/dev/athn/headers/athnvar.h [hunk 1]
```diff
@@ -628,3 +630,4 @@ extern void	athn_detach(struct athn_softc *);
 extern void	athn_suspend(struct athn_softc *);
 extern void	athn_wakeup(struct athn_softc *);
 extern int	athn_intr(void *);
+#endif
```
Comment (nzulinski-mobica) [inline sys/dev/athn/headers/athnvar.h]: Please add empty lina at the end of file

File: sys/dev/athn/headers/if_athn_usb.h [hunk 2]
```diff
@@ -515,3 +526,5 @@ struct athn_usb_softc {
 					    struct ieee80211_node *);
 	int				sc_key_tasks;
 };
+
+#endif
```
Comment (nzulinski-mobica) [inline sys/dev/athn/headers/if_athn_usb.h]: ditto

File: sys/dev/athn/if_athn_usb.c [hunk 3]
```diff
@@ -593,6 +641,10 @@ athn_usb_attachhook(device_t self)
 	if (error != 0)
 		return;
 
+	error = athn_usb_verify_fw_ver(sc, &img_ver);
```
Comment (nzulinski-mobica) [inline sys/dev/athn/if_athn_usb.c:650]: shouldn't it be before athn_usb_load_firmware? Because now we are checking if version of firmware that we loaded

**Comment by Mikolaj-Filar-Mobica** [comment]
This PR needs to be thoroughly reviewed by Norbert, as it's unknown why the OpenBSD code needed these changes. Since this PR isn't critical now, it can wait for someone with deeper knowledge of USB.

**Review by nzulinski-mobica — APPROVED** [review]

**Pull request merged**